sumpy/e2p.py

```diff
@@ -81,15 +84,18 @@ def __init__(self, ctx, expansion, kernels,
     def default_name(self):
         pass
 
+    @memoize_method
+    def get_cached_loopy_knl_and_optimizations(self):
```

Comment:
I don't like the naming of this, there's already `get_kernel`. The name should reflect that we're talking about the evaluator. Plus whether or not something is memoized isn't typically reflected in its name.

Reply:
Fixed now.